packages/clay-tooltip/src/TooltipProvider.tsx

```diff
@@ -231,10 +195,25 @@ const TooltipProvider: React.FunctionComponent<{
 		}
 	}, [align, show]);
 
-	warning(
-		children.type !== React.Fragment,
-		'<TooltipProvider />: React Fragment is not allowed as a child to TooltipProvider. Child must be a single HTML element.'
-	);
+	useEffect(() => {
+		if (!(window as any)[SYMBOL_TOOLTIP]) {
+			(window as any)[SYMBOL_TOOLTIP] = true;
```

Comment:
These `any`s scare me a bit. I'm also not sure the best way to handle global window values....

Reply:
You can augment the `Window` interface as described in [some of the answers here](https://stackoverflow.com/questions/12709074/how-do-you-explicitly-set-a-new-property-on-window-in-typescript) but having done this in several projects and several versions of TS, I've always found it to be a tremendous pain, incredibly delicate, and sensitive to versions, config settings, project structure etc. Basically, it's really hard, and I suspect (wild speculation) that one of the reasons they don't particularly prioritize making it easy is because they don't want to encourage people spewing stuff into the global scope. Just a theory though.

Here's my most recent example of fighting with this, where I [add `Liferay.State`](https://github.com/liferay/liferay-portal/blob/4b4ff53717645e7bf4c4b373b5cb5d218fb1207a/modules/apps/frontend-js/frontend-js-state-web/src/main/resources/META-INF/resources/State.ts#L510-L524) to the global `window` object in DXP. @bryceosterhaus, you are probably going to have to take this kind of thing into consideration when you do [LPS-131257](https://issues.liferay.com/browse/LPS-131257).

Reply:
Yeah, I didn't even get to extend the `Window` interface, it's what I normally do in a NodeJS environment for `process.env`. But we can try to do the same thing @wincent did for State. Probably just adding to the same file or creating a `window.d.ts`?

Reply:
So in `@liferay/frontend-js-state-web` I initially tried to declare all this as an ambient type in a separate file, with a view towards centralizing it in the future, but then the whole project references thing caused things to blow up, requiring the nonsense that I ended up stick down the bottom of that file. Because of the sensitivity to versions, config, and structure, there is no guarantee that a technique you see working in one place will work when transplanted to another. 😢